src/main/java/com/sap/sgs/phosphor/fosstars/data/github/NvdEntryMatcher.java

```diff
@@ -209,63 +234,45 @@ private static boolean longestCommonSubsequenceCheck(String one, String two) {
    *      Similarity</a>
    */
   private static boolean jaroWinklerSimilarityCheck(String one, String two) {
-    return JWS.apply(one, two) > METRIC_THRESHOLD;
+    return JWS.apply(one, two) > JWS_METRIC_THRESHOLD;
   }
 
   /**
-   * Checks if one of the cpeXXUri contains the project's name.
+   * Checks if one of the cpeXXUri contains the project's and organization's names. If the check is
+   * a success, it returns true immediately. If it could not match the organization's name then
+   * check if the list of references contains any information.
    * 
    * @param cpeMatch of type {@link CpeMatch}.
+   * @param List of references of type {@link ReferenceLink}.
    * @return True if one of the cpeXXUri contains the project's name. Otherwise False.
    */
-  private boolean cpeUriCheck(CpeMatch cpeMatch) {
-    return match(productFrom23Uri(cpeMatch.getCpe23Uri()))
-        || match(productFrom22Uri(cpeMatch.getCpe22Uri()));
-  }
+  private boolean projectCheck(CpeMatch cpeMatch, List<ReferenceLink> references) {
+    CpeUri cpeUri = cpeMatch.getCpeUri();
 
-  /**
-   * Returns product name from {@link CpeMatch#getCpe23Uri()}.
-   */
-  private static String productFrom23Uri(String cpe23Uri) {
-    return stringAtIndex(cpe23Uri, ":", 4);
-  }
+    boolean productMatch = match(cpeUri.getProduct(), project.name());
+    boolean vendorMatch = match(cpeUri.getVendor(), project.organization().name());
 
-  /**
-   * Returns product name from {@link CpeMatch#getCpe22Uri()}.
-   */
-  private static String productFrom22Uri(String cpe22Uri) {
-    return stringAtIndex(cpe22Uri, ":", 3);
-  }
+    if (productMatch && vendorMatch) {
+      return true;
+    }
 
-  /**
-   * Returns the substring present at the index after the split by the given delimiter pattern.
-   * 
-   * @param content The String to be split.
-   * @param pattern The delimiter to be used.
-   * @param index The index of the sub-string which should be returned.
-   * @return The sub-string if the index is valid and the content is not null. Otherwise returns an
-   *         empty string.
-   */
-  private static String stringAtIndex(String content, String pattern, int index) {
-    if (content != null) {
-      String[] splitContent = content.split(pattern);
-      if (splitContent.length > index) {
-        return splitContent[index];
-      }
+    boolean referenceMatch = match(references);
+    if (productMatch || referenceMatch) {
```

Comment:
Why does it use OR? Should it be AND? Or, would it make sense to use `(productMatch || vendorMatch) && referenceMatch`?